lib/index.js

```diff
@@ -271,12 +307,36 @@ api.update = function(actor, identity, options, callback) {
     };
   }
   */
-  async.waterfall([
-    function(callback) {
-      api.checkPermission(
-        actor, PERMISSIONS.IDENTITY_EDIT, {resource: identity}, callback);
+  async.auto({
+    get: callback => {
+      const query = {
+        id: database.hash(identity.id),
+        'identity.sysStatus': options.deleted ? 'deleted' : 'active'
+      };
+      database.collections.identity.findOne(query, {
+        'identity.id': 1,
+        'identity.owner': 1
+      }, (err, record) => callback(err, record ? record.identity : null));
     },
-    function(callback) {
+    checkPermission: ['get', (callback, results) => {
+      const resources = [identity.id];
+      if(results.get && results.get.owner) {
+        resources.push(results.get.owner);
+      }
+      api.checkPermission(
+        actor, PERMISSIONS.IDENTITY_EDIT, {resource: resources}, callback);
+    }],
+    checkMemberOf: ['checkPermission', (callback, results) => {
+      if(identity.memberOf) {
+        if(!Array.isArray(identity.memberOf)) {
+          identity.memberOf = [identity.memberOf];
```

Comment:
Can we just do a lodash 'make all entries unique' call instead of checking? Clean up the data automatically instead of rejecting since we know what to do in the case where items aren't unique?